packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -203,6 +206,13 @@ Future<Map<String, dynamic>> _getAllCoverage(VMService service, bool Function(St
     final Map<String, Map<String, dynamic>> sourceReports = <String, Map<String, dynamic>>{};
     // For each ScriptRef loaded into the VM, load the corresponding Script and
     // SourceReport object.
+
+    // We may receive such objects as
+    // {type: Sentinel, kind: Collected, valueAsString: <collected>}
```

Comment:
@bkonyi why does the getScripts vmservice rpc return this?

Reply:
As far as I can tell it should never return anything but a `ScriptList`. I can't imagine a situation where a sentinel would ever be returned (see implementation [here](https://github.com/dart-lang/sdk/blob/master/runtime/vm/service.cc#L1404)).

Reply:
Maybe the request is referring to an object that was garbage collected, and that is caught before the implementation is invoked? Is there any way to know which argument was collected? @jonahwilliams is it possible the getScripts call could be referring to something stale?

Reply:
`getScripts` only takes an isolate ID as a parameter, so maybe an isolate had been shutdown? I'm not sure what the service does for isolate specific requests when the isolate has already shutdown, but returning a sentinel seems possible in that case.

Reply:
I'm not really sure how that would happen, or how it hasn't happened yet when we run coverage. The repro project in #35120 fairly trivial though